pkg/api/universal/conversation_stream.go

```diff
@@ -0,0 +1,312 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package universal
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	piiscrubber "github.com/aavaz-ai/pii-scrubber"
+
+	"github.com/dapr/components-contrib/conversation"
+	diag "github.com/dapr/dapr/pkg/diagnostics"
+	"github.com/dapr/dapr/pkg/messages"
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/kit/logger"
+	kmeta "github.com/dapr/kit/metadata"
+)
+
+// defaultPIIScrubberWindowSize is the default window size (bytes) for PII scrubbing
+// this allows for a look-ahead window of stream data to be scrubbed
+// TODO: make this configurable
+const defaultPIIScrubberWindowSize = 50
+
+// ConverseStreamAlpha1 implements the streaming conversation API
+func (a *Universal) ConverseStreamAlpha1(req *runtimev1pb.ConversationRequest, stream runtimev1pb.Dapr_ConverseStreamAlpha1Server) (retErr error) {
+	// Validate component exists
+	if a.compStore.ConversationsLen() == 0 {
+		err := messages.ErrConversationNotFound
+		a.logger.Debug(err)
+		return err
+	}
+
+	component, ok := a.compStore.GetConversation(req.GetName())
+	if !ok {
+		err := messages.ErrConversationNotFound.WithFormat(req.GetName())
+		a.logger.Debug(err)
+		return err
+	}
+
+	// Prepare component conversation request (similar to non-streaming version)
+	request := &conversation.ConversationRequest{}
+	err := kmeta.DecodeMetadata(req.GetMetadata(), request)
+	if err != nil {
+		return err
+	}
+
+	if len(req.GetInputs()) == 0 {
+		err := messages.ErrConversationMissingInputs.WithFormat(req.GetName())
+		a.logger.Debug(err)
+		return err
+	}
+
+	// Create streaming pipeline
+	pipelineImpl := NewStreamingPipelineImpl(a.logger)
+
+	// prepare scrubber in case PII scrubbing is enabled
+	var scrubber piiscrubber.Scrubber
+
+	// Check if PII scrubbing is needed for inputs or outputs
+	needsInputPIIScrubbing := needsInputScrubber(req)
+	needsOutputPIIScrubbing := needsOutputScrubber(req)
+
+	if needsInputPIIScrubbing || needsOutputPIIScrubbing {
+		scrubber, err = piiscrubber.NewDefaultScrubber()
+		if err != nil {
+			err := messages.ErrConversationMissingInputs.WithFormat(req.GetName())
+			a.logger.Debug(err)
+			return err
+		}
+
+		// Add PII scrubbing middleware for streaming output scrubbing
+		if needsOutputPIIScrubbing {
+			scrubberMiddleware, scrubberErr := NewStreamingPIIScrubber(defaultPIIScrubberWindowSize)
+			if scrubberErr != nil {
+				return fmt.Errorf("failed to create streaming PII scrubber middleware: %w", scrubberErr)
+			}
+			pipelineImpl.AddMiddleware(scrubberMiddleware)
```

Comment:
Why is this being added as a middleware in this way if it is always going to be added?